### PR TITLE
Fix Base.show for SSHManager and LocalManager

### LINKED ISSUE #34933: Unreached methods for SSHManager and LocalManager
I wanted to add a few simple tests for 

https://coveralls.io/builds/29043355/source?filename=stdlib/Distributed/src/managers.jl#L147
https://coveralls.io/builds/29043355/source?filename=stdlib/Distributed/src/managers.jl#L369

But it seems these methods are never dispached to. I also tried `invoke(show, Tuple{IO, SSHManager}, b, SSHManager("127.0.0.1"))` without success.

Should these methods be removed? Why doesn't invoke work here?

### PR DESCRIPTION
**Pull request by carstenbauer**
See https://github.com/JuliaLang/julia/issues/34933.

This PR makes the current `show` methods in `Distributed` for `LocalManager` and `SSHManager` extend `Base.show` (seems to have been a mistake before). Also added two tests to check that they actually get called now.

Closes #34933

### CODE REVIEW

File: stdlib/Distributed/src/managers.jl [hunk 1]
```diff
@@ -144,7 +144,7 @@ function launch(manager::SSHManager, params::Dict, launched::Array, launch_ntfy:
 end
 
 
-show(io::IO, manager::SSHManager) = println(io, "SSHManager(machines=", manager.machines, ")")
+Base.show(io::IO, manager::SSHManager) = println(io, "SSHManager(machines=", manager.machines, ")")
```
Comment (fredrikekre) [inline stdlib/Distributed/src/managers.jl]:
```suggestion
Base.show(io::IO, manager::SSHManager) = print(io, "SSHManager(machines=", manager.machines, ")")
```

File: stdlib/Distributed/src/managers.jl [hunk 2]
```diff
@@ -366,7 +366,7 @@ function addprocs(np::Integer; restrict=true, kwargs...)
     addprocs(LocalManager(np, restrict); kwargs...)
 end
 
-show(io::IO, manager::LocalManager) = println(io, "LocalManager()")
+Base.show(io::IO, manager::LocalManager) = println(io, "LocalManager()")
```
Comment (fredrikekre) [inline stdlib/Distributed/src/managers.jl]:
```suggestion
Base.show(io::IO, manager::LocalManager) = print(io, "LocalManager()")
```

File: stdlib/Distributed/test/managers.jl [hunk 3]
```diff
@@ -22,3 +22,6 @@ addr, port = getsockname(sock)
 sock = bind_client_port(TCPSocket(), typeof(IPv6(0)))
 addr, port = getsockname(sock)
 @test addr == ip"::"
+
+@test sprint((t,x) -> show(t, "text/plain", x), SSHManager("127.0.0.1")) == "SSHManager(machines=Dict{Any,Any}('.' => 3,'0' => 2,'1' => 2,'2' => 1,'7' => 1))\n"
```
Comment (fredrikekre) [inline stdlib/Distributed/test/managers.jl]:
```suggestion
@test sprint((t,x) -> show(t, "text/plain", x), SSHManager("127.0.0.1")) == "SSHManager(machines=Dict{Any,Any}('.' => 3,'0' => 2,'1' => 2,'2' => 1,'7' => 1))"
```

File: stdlib/Distributed/test/managers.jl [hunk 4]
```diff
@@ -22,3 +22,6 @@ addr, port = getsockname(sock)
...
+@test sprint((t,x) -> show(t, "text/plain", x), LocalManager(1, true)) == "LocalManager()\n"
```
Comment (fredrikekre) [inline stdlib/Distributed/test/managers.jl]:
```suggestion
@test sprint((t,x) -> show(t, "text/plain", x), LocalManager(1, true)) == "LocalManager()"
```

**Comment by crstnbr** [comment]
Apparently the order of the dict entries varies across different CI runs. Any idea how to test the `show` output in this case? Perhaps it's enough to check everything but the `Dict` printing, that is only the prefix "SSHManager(machines=" and the trailing parenthesis?

File: stdlib/Distributed/test/managers.jl [hunk 5]
```diff
@@ -22,3 +22,6 @@ addr, port = getsockname(sock)
 sock = bind_client_port(TCPSocket(), typeof(IPv6(0)))
 addr, port = getsockname(sock)
 @test addr == ip"::"
+
+@test sprint((t,x) -> show(t, "text/plain", x), SSHManager("127.0.0.1")) == "SSHManager(machines=Dict{Any,Any}('.' => 3,'0' => 2,'1' => 2,'2' => 1,'7' => 1))"
```
Comment (fredrikekre) [inline stdlib/Distributed/test/managers.jl]:
```suggestion
@test occursin(r"^SSHManager\(machines=.*\)$",
               sprint((t,x) -> show(t, "text/plain", x), SSHManager("127.0.0.1")))
```

**Pull request merged**